Build Updates

rossmcdonald opened the pull request:
- [x] CHANGELOG.md updated
- [X] Rebased/mergable
- [x] Tests pass
- [X] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

---
Updates here: 
- Improved handling of tar and zip package outputs
- Added support for statically-compiled binary outputs
- Changed the nightly versioning convention from the Unix epoch to UTC `YYYYMMDDhhmm`

Fixes #5554 #5252

jsternberg commented:
LGTM but you may want to have someone with more knowledge of the build script look at it too.

rossmcdonald commented:
/cc @nathanielc This PR this fixes issues with zips and tarballs, which will be ported to Kapacitor assuming everything looks okay.

nathanielc commented on build.py:
```diff
@@ -109,16 +109,20 @@ def create_package_fs(build_root):
         os.chmod(os.path.join(build_root, d), 0755)
 
-def package_scripts(build_root):
+def package_scripts(build_root, static=False):
```
So if I understand correctly by passing static here we don't copy all the service scripts etc just the config. Then for builds marked static and all windows build we set static=True since these extra scripts don't apply.

The name `static` for this method is confusing maybe we should name this parameter `config_only`?

Then this code below dosen't read as all windows builds are static:

```python
if platform == 'windows' or static:
    package_scripts(build_root, static=True)
```

It is a simple fix but it was quite confusing at first. 

Also why do static build not need the extra service scripts etc? Is it because they are intended for use in apline linux? If so a comment explaining that would be helpful.

rossmcdonald replied:
> Also why do static build not need the extra service scripts etc? Is it because they are intended for use in apline linux? If so a comment explaining that would be helpful.

That is correct. The only people who should be using the static builds are people who just need binaries, so Alpine users and Windows users are the first ones who come to mind. Do you think all tars and zips should be this way?

> The name static for this method is confusing maybe we should name this parameter config_only?

Agreed. I'll change it. I was originally thinking of 'static' packages (just plain zip, tar), but I can see how it's misleading and/or vague.

nathanielc replied:
> Do you think all tars and zips should be this way?

No, I think it is useful to provide the scripts in tars, zips since they have value, at least in the linux versions. To me the contents of a package (tar, deb, zip etc)  depends on the platform and not the package type. No matter the package type provide the relevant content for the platform. 

Why do Windows users want static builds? I get why they don't care about service scripts etc but not static builds.

rossmcdonald replied:
> Why do Windows users want static builds?

Sorry, Windows builds won't be statically compiled, but they will use the `static` flag (soon to be ` config_only`) for the packaging portion. Another reason why `static` should not have been used as the flag name. :smile:

rossmcdonald commented:
@nathanielc I've changed the `static` flag for `package_scripts` to `config_only` for clarification.

nathanielc commented:
@rossmcdonald Looks good.

rossmcdonald commented:
Rebased.

@nathanielc I also added one final change to the nightly date format. The format is now UTC `%Y%m%d%H%M` (or `YYYYMMDDhhmm`) over the unix epoch.
